src/QSymbolicsBase/basic_ops_homogeneous.jl

```diff
@@ -31,10 +30,14 @@ operation(x::SScaled) = *
 head(x::SScaled) = :*
 children(x::SScaled) = [:*,x.coeff,x.obj]
 function Base.:(*)(c, x::Symbolic{T}) where {T<:QObj} 
-    if iszero(c) || iszero(x)
+    if (isa(c, Number) && iszero(c)) || iszero(x)
```

Comment:
This change fixes the type piracy issue with `iszero`. Now the `*` call checks if the coefficient is a number (and not a symbolic complex number such as an inner product) before it checks if it is zero.